wagtail/core/blocks/field_block.py

```diff
@@ -91,6 +91,10 @@ def js_args(self, block, context):
             },
         ]
 
+    def get_media(self, block, context):
+        media = super().get_media(block, context)
+        return media + block.field.widget.media
```

Comment:
I think this should happen in the widget's own adapter rather than FieldBlock's. Handily you've just implemented [the TODO I left there](https://github.com/wagtail/wagtail/blob/f6f06175e0362d04140eaa6c4972c9e9bf8d0a21/wagtail/core/widget_adapters.py#L23-L25), so that should be a straightforward fix :-)

Reply:
Oh nice! Forgot widgets had their own adapters!

Reply:
Fixed! (squashed into the last commit)